TestC2NDebugNetmap flake

https://github.com/tailscale/tailscale/actions/runs/18046052951/job/51356963396?pr=17296

```
=== RUN   TestC2NDebugNetmap
=== PAUSE TestC2NDebugNetmap
=== CONT  TestC2NDebugNetmap
    integration.go:528: DERP httpsrv listener: 127.0.0.1:40931
    integration.go:562: control URL: http://127.0.0.1:46393/
    integration.go:609: 2025/09/26 18:27:46 DEBUG-ADDR=127.0.0.1:43561
    integration_test.go:1796: Status: NeedsLogin
    integration_test.go:1797: Running /tmp/TestC2NDebugNetmap2443331146/001/tailscale --socket=/tmp/TestC2NDebugNetmap2443331146/002/tailscale.sock up --login-server=http://127.0.0.1:46393/ --reset ...
    logger.go:105: testcontrol: Created user userid:1 for node nodekey:f9f518bd3f965062454db878fe3b4b0c7986f3f2ebebe86f4db9f5015cf54e0a
    integration.go:609: 2025/09/26 18:27:46 DEBUG-ADDR=127.0.0.1:37083
    integration_test.go:1796: Status: NeedsLogin
    integration_test.go:1797: Running /tmp/TestC2NDebugNetmap2443331146/001/tailscale --socket=/tmp/TestC2NDebugNetmap2443331146/008/tailscale.sock up --login-server=http://127.0.0.1:46393/ --reset ...
    logger.go:105: testcontrol: Created user userid:2 for node nodekey:99c7ab3899a0c735f46925b170158ed25d0cbfb1e2bfe6edc5f52baf67170978
    integration_test.go:1864: Peers differ (-current +candidate):
          []tailcfg.NodeView{
        - 	{
        - 		ж: &tailcfg.Node{
        - 			ID:       1,
        - 			StableID: "TESTCTRL00000001",
        - 			Name:     "runnervm3ublj",
        - 			User:     1,
        - 			Key: key.NodePublic{
        - 				k: [32]uint8{
        - 					0xf9, 0xf5, 0x18, 0xbd, 0x3f, 0x96, 0x50, 0x62, 0x45, 0x4d, 0xb8, 0x78, 0xfe,
        - 					0x3b, 0x4b, 0x0c, ...,
        + 					},
        + 					port: 56688,
        + 				},
        + 			},
        + 			HomeDERP: 1,
        + 			Hostinfo: tailcfg.HostinfoView{
        + 				ж: &tailcfg.Hostinfo{
        + 					IPNVersion:     "1.89.0-dev20250926-t6fed5d05c",
        + 					BackendLogID:   "30f9187d43f843e55ab5dec30502a09717c2188a3eebf0467a449ddeff8d666d",
        + 					OS:             "linux",
        + 					OSVersion:      "6.11.0-1018-azure",
        + 					Container:      "false",
        + 					Distro:         "ubuntu",
        + 					DistroVersion:  "24.04",
        + 					DistroCodeName: "noble",
        + 					Desktop:        "false",
        + 					Hostname:       "runnervm3ublj",
        + 					Machine:        "x86_64",
        + 					GoArch:         "386",
        + 					GoArchVar:      "sse2",
        + 					GoVersion:      "go1.25.1",
        + 					Services: []tailcfg.Service{
        + 						{Proto: "tcp", Port: 22}, {Proto: "tcp", Port: 53},
        + 						{Proto: "tcp", Port: 34152, Description: "tsconsensus.test"},
        + 						{Proto: "tcp", Port: 35751, Description: "tsconsensus.test"},
        + 						{Proto: "tcp", Port: 36467, Description: "tsconsensus.test"},
        + 						{Proto: "tcp", Port: 40226, Description: "tsnet.test"},
        + 						{Proto: "tcp", Port: 42739, Description: "tsconsensus.test"},
        + 						{Proto: "tcp", Port: 46590, Description: "tsconsensus.test"}, ...,
        + 					},
        + 					NetInfo: &tailcfg.NetInfo{
        + 						MappingVariesByDestIP: "false",
        + 						HairPinning:           "unset",
        + 						WorkingIPv6:           "false",
        + 						OSHasIPv6:             "true",
        + 						WorkingUDP:            "true",
        + 						WorkingICMPv4:         "false",
        + 						UPnP:                  "false",
        + 						PMP:                   "false",
        + 						...
        + 					},
        + 					...
        + 				},
        + 			},
        + 			Cap:               127,
        + 			MachineAuthorized: true,
        + 			Capabilities: []tailcfg.NodeCapability{
        + 				"https", "funnel", "https://tailscale.com/cap/file-sharing",
        + 				"https://tailscale.com/cap/funnel-ports?ports=8080,443",
        + 			},
        + 			CapMap: tailcfg.NodeCapMap{
        + 				"funnel":                                                {},
        + 				"https":                                                 {},
        + 				"https://tailscale.com/cap/file-sharing":                {},
        + 				"https://tailscale.com/cap/funnel-ports?ports=8080,443": {},
        + 			},
        + 			ComputedName:         "runnervm3ublj",
        + 			ComputedNameWithHost: "runnervm3ublj",
        + 		},
        + 	},
          }
    stuntest.go:65: STUN server shutdown
--- FAIL: TestC2NDebugNetmap (1.48s)

```

cc @knyar

--- a/tstest/integration/integration_test.go
+++ b/tstest/integration/integration_test.go
@@ -1784,7 +1784,9 @@ func TestPeerRelayPing(t *testing.T) {
 func TestC2NDebugNetmap(t *testing.T) {
 	tstest.Shard(t)
 	tstest.Parallel(t)
-	env := NewTestEnv(t)
+	env := NewTestEnv(t, ConfigureControl(func(s *testcontrol.Server) {
+		s.CollectServices = "false"
+	}))
 
 	var testNodes []*TestNode
 	var nodes []*tailcfg.Node

--- a/tstest/integration/testcontrol/testcontrol.go
+++ b/tstest/integration/testcontrol/testcontrol.go
@@ -35,6 +35,7 @@ import (
 	"tailscale.com/tailcfg"
 	"tailscale.com/types/key"
 	"tailscale.com/types/logger"
+	"tailscale.com/types/opt"
 	"tailscale.com/types/ptr"
 	"tailscale.com/util/httpm"
 	"tailscale.com/util/mak"
@@ -68,6 +69,10 @@ type Server struct {
 
 	// DefaultNodeCapabilities overrides the capability map sent to each client.
 	DefaultNodeCapabilities *tailcfg.NodeCapMap
+
+	// CollectServices, if non-empty, sets whether the control server asks
+	// for service updates. If empty, the default is "true".
+	CollectServices opt.Bool
 
 	// ExplicitBaseURL or HTTPTestServer must be set.
 	ExplicitBaseURL string           // e.g. "http://127.0.0.1:1234" with no trailing URL
@@ -1096,7 +1101,7 @@ func (s *Server) MapResponse(req *tailcfg.MapRequest) (res *tailcfg.MapResponse,
 		Node:            node,
 		DERPMap:         s.DERPMap,
 		Domain:          domain,
-		CollectServices: "true",
+		CollectServices: cmp.Or(s.CollectServices, "true"),
 		PacketFilter:    packetFilterWithIngress(s.PeerRelayGrants),
 		DNSConfig:       dns,
 		ControlTime:     &t,

